powerauth-java-crypto/src/main/java/io/getlime/security/powerauth/crypto/lib/v4/kdf/Kdf.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * PowerAuth Crypto Library
+ * Copyright 2024 Wultra s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.getlime.security.powerauth.crypto.lib.v4.kdf;
+
+import io.getlime.security.powerauth.crypto.lib.model.exception.GenericCryptoException;
+import io.getlime.security.powerauth.crypto.lib.util.ByteUtils;
+import io.getlime.security.powerauth.crypto.lib.util.KeyConvertor;
+import org.bouncycastle.crypto.macs.KMAC;
+import org.bouncycastle.crypto.params.KeyParameter;
+
+import javax.crypto.SecretKey;
+import java.nio.charset.StandardCharsets;
+
+/**
+ * Universal KDF based on KMAC-256 (Keccak).
+ *
+ * @author Roman Strobl, roman.strobl@wultra.com
+ */
+public class Kdf {
+
+    private static final String CRYPTO4_KDF_CUSTOM_STRING = "PA4KDF";
+    private static final Integer KMAC_ALGORITHM_STRENGTH = 256;
```

Comment:
This is probably the length or size, not "strength"?

Reply:
See:
https://nvlpubs.nist.gov/nistpubs/specialpublications/nist.sp.800-185.pdf

Similarly, cSHAKE256, KMAC256, TupleHash256, and
ParallelHash256 each provides a security strength of 256 bits against generic attacks.

I can use `KMAC_ALGORITHM_STRENGTH_BITS` if you prefer?

Reply:
Renamed to `KMAC_BIT_LENGTH`.